docs/spec/interfaces.md

```diff
@@ -0,0 +1,93 @@
+# Interface Contracts
+
+<!-- This image is sourced from https://drive.google.com/open?id=1Mgn0Oz85M9W1Pwfnw3u0OhZmIIxSxcM3lZUEZyPCoB0 -->
+![Interface Contracts Overview](images/interface-contracts-overview.svg)
+
+---
+
+## Subscribable
+
+A **Subscribable** resource will emit events that a _Subscription_ can direct
+to a _Targetable_ or _Sinkable_ resource.
+
+### Control Plane
+
+A **Subscribable** resource may be referenced in the _from_ section of a
+_Subscription_. The **Subscribable** resource MUST expose a
+_subscribable.channelable_ field (an ObjectReference) in its _status_ section.
+This _channelable_ field MUST be _Channelable_, and _channel_ MAY refer back to
+the **Subscribable** resource.
+
+### Data Plane
+
+A **Subscribable** resource is an event producer or forwarder, events that it
+produces or forwards are delivered via its _status.channel_ resource.
+
```

Comment:
Is it worth enumerating which resources can act as which types?

Reply:
I think the more important concept is how the current objects map to the interfaces.  When we have more than one implementation of the interfaces then we can look showing which known impl for an interface is available, like the java docs tend to do.

Reply:
I think it would be worth documenting _somewhere_ (possibly not in the interface spec) which types are **known** to implement which interfaces. Finding the implementations of an interface tends to be a common activity that people do when learning about a codebase.